Assets/Simon/Scripts/Sequence.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+public class Sequence : MonoBehaviour
+{
+    public List<Colors> ColorSequence { get; private set; }
+
+    private int sequenceIndex = 0;
+    public void CreateColorSequence(int sequenceLength)
+    {
+        if(sequenceLength <= 0)
+        {
+            Debug.LogError("The length of the sequence should be larger than 0");
+        }
+
+        ColorSequence = new List<Colors>();
+
+        for(int i = 0; i < sequenceLength; i++)
+        {
+            ExtendColorSequence();
+        }
+    }
+
+    public void IterateSequence()
+    {
+       sequenceIndex++;
+    }
+
+    public bool SequenceReachedLastItem()

```

Comment:
I would use a property personally as it doesn't accept any parameters.

```public bool SequenceReachedLastItem => sequenceIndex >= ColorSequence.Count;```

Reply:
Why is a property better than a method here? Because it is more compact?

Reply:
I think because it's more like state than behaviour. That's my opinion anyway.

i.e. it's closer to data, rather than some operation. It has to do a comparison but to me it makes more sense if it reads like:

```if (SomeState) { } ```. rather than ```if(SomeOperation()) { }```

Reply:
I see. It is a very simple comparison rather than an operation. No data is manipulated. There's no algorithm running.